app/src/main/java/com/vero/tasky/agenda/data/remote/repository/AgendaRepositoryImpl.kt

```diff
@@ -18,25 +24,65 @@ class AgendaRepositoryImpl(
     private val reminderDao: ReminderDao,
 ) : AgendaRepository {
 
-    override suspend fun getAgendaForDay(timezone: String, time: Long): Flow<Result<Agenda>> = flow {
+    override suspend fun getAgendaForDay(
+        timezone: String,
+        timestamp: Long
+    ) : Flow<List<AgendaItem>> = flow {
+
+        emit(getAgendaItemsFromDao(timestamp))
+
+        ApiCallHandler.safeApiCall {
+            val networkAgendaItems = api.getAgendaForDay(timezone, timestamp)
+            saveAgendaItems(networkAgendaItems)
+        }
+
+        emit(getAgendaItemsFromDao(timestamp))
     }
 
     override suspend fun syncAgenda(
         deletedEventsIds: List<String>,
         deletedTasksIds: List<String>,
         deletedReminderIds: List<String>
     ): Result<Unit> = ApiCallHandler.safeApiCall {
-            api.syncAgenda(
-                SyncAgendaRequest(
-                    deletedEventsIds = deletedEventsIds,
-                    deletedTasksIds = deletedTasksIds,
-                    deletedReminderIds = deletedReminderIds
-                )
+        api.syncAgenda(
+            SyncAgendaRequest(
+                deletedEventsIds = deletedEventsIds,
+                deletedTasksIds = deletedTasksIds,
+                deletedReminderIds = deletedReminderIds
             )
+        )
     }
 
-    override suspend fun getFullAgenda() =
-        ApiCallHandler.safeApiCall {
-            Agenda(emptyList())
+    override suspend fun getFullAgenda() = ApiCallHandler.safeApiCall {
```

Comment:
Also for this `ApiCallHandler` I think the `object` is rather unnecessary and you could put the `safeApiCall` function just in a plain Kotlin file, since it's not more clear to me what's inside this object if I read its name (rather less clear 😅 )